File: leetcode2026/2026_02/leetcode_3634_minRemoval.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int r = 0;
        int res = n-1;
        for (int l=0; l<n; l++){
            while (r<n && 1ll*nums[r] <= 1ll*nums[l]*k){
                r++;
            }
            res = min(res, n-(r-l));
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {2,1,5};
    int k = 2;
    cout << s.minRemoval(nums, k) << endl;
}

File: leetcode2026/2026_02/leetcode_3634_minRemoval.py
from typing import List

class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)
        r = 0
        res = n-1
        for l in range(n):
            while r < n and nums[r] <= nums[l]*k:
                r += 1
            cur = n-(r-l)
            if cur < res:
                res = cur
        return res

        # n = len(nums)
        # nums.sort()
        # res = float("inf")
        # for i in range(n):
        #     res = min(res, i+n-1-(bisect_right(nums, nums[i]*k)-1))
        #     res = min(res, (n-1-i)+bisect_left(nums, (nums[i]+k-1)//k))
        # return res

if __name__ == "__main__":
    s = Solution()
    nums = [2,1,5]
    k = 2
    print(s.minRemoval(nums, k))
